api/app/src/mappings/garmin/user.ts

```diff
@@ -0,0 +1,64 @@
+import { Biometrics } from "@metriport/api";
+import { groupBy } from "lodash";
+import { z } from "zod";
+import { DataType, garminMetaSchema, garminTypes, User, UserData } from ".";
+import { PROVIDER_GARMIN } from "../../shared/constants";
+
+export const mapToBiometricsFromUser = (
+  items: GarminUserMetricsList
+): UserData<Biometrics>[] => {
+  const type: DataType = "biometrics";
+  const itemsByUAT = groupBy(items, (a) => a.userAccessToken);
+  return Object.entries(itemsByUAT).flatMap(([key, values]) => {
+    const uat = key;
+    const userData = values;
+    const user: User = {
+      userAccessToken: uat,
+    };
+    const mappedItems = userData.map(garminUserMetricsToBody);
+    const definedItems: Biometrics[] = mappedItems.filter(
+      (v: Biometrics | undefined) => v != undefined
+    ) as Biometrics[];
+    return definedItems.map((data) => ({
+      user,
+      typedData: { type, data },
+    }));
+  });
+};
+
+export const garminUserMetricsToBody = (
+  gBody: GarminUserMetrics
+): Biometrics | undefined => {
+  const bio: Biometrics = {
+    metadata: {
+      date: gBody.calendarDate,
```

Comment:
Should we use the query date to guarantee this is `YYYY-MM-DD`, or is this not an issue?

Reply:
This comes from Garmin, I managed to add some validation with this + dayjs' `CustomParseFormat` ([source](https://day.js.org/docs/en/parse/is-valid)):

`  date: z.string().refine((v) => dayjs(v, ISO_DATE, true).isValid()),`

Reply:
Ended up adding this on this upstream PR's [commit](https://github.com/metriport/metriport/pull/18/commits/8809ba9ee2bd819015166899b6a3729e95538657).